torch_brain/models/poyo_plus/callbacks.py

```
@@ -0,0 +1,508 @@
+                or metric_factory must be provided.
+            metric_factory: (Optional) A callable that returns an instance of the metric to be used.
+                If not provided, the metric is inferred based on the modality_spec.
+            quiet: If True, disables the logging of the metrics to the console.
+        """
+        _check_lightning_available(self.__class__)
+        self.quiet = quiet
+
+        if metric_factory is not None:
+            pass
+        elif modality_spec.type == DataType.CONTINUOUS:
+            metric_factory = lambda: torchmetrics.R2Score()
```

Ruff (0.15.13)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `metric_factory` as a `def`

(E731)